src/plugins/intel_cpu/src/nodes/interaction.cpp

```diff
@@ -0,0 +1,220 @@
+            LayoutType::ncsp,
+            dataPrecision,
+            getOutputShapeAtPort(0),
+            false,
+            -1
+        }
+    };
+    //add descriptor
+    addSupportedPrimDesc(inPortConfigs, outPortConfigs, impl_desc_type::ref_any, true);
+}
+
+static inline void cat(const uint8_t* in1, const uint8_t* in2, uint8_t* out,
+    size_t in1Size, size_t in2Size, size_t elemSize) {
+    cpu_memcpy(out, in1, in1Size * elemSize);
+    cpu_memcpy(out + in1Size * elemSize, in2, in2Size * elemSize);
+}
+
+static inline void cat(uint8_t* out,
+                       const std::vector<const uint8_t*>& in,
+                       const std::vector<uint32_t>& feature_sizes,
+                       int64_t bs,
+                       size_t elemSize) {
+    size_t offset = 0;
+    for (int j = 0; j < feature_sizes.size(); j++) {
+        cpu_memcpy(out + offset * elemSize, in[j] + bs * feature_sizes[j] * elemSize,
+            feature_sizes[j] * elemSize);
+        offset += feature_sizes[j];
+    }
+}
+
+
+static inline void flat_triangle(const uint8_t* in, uint8_t* out, size_t size, size_t elemSize) {
+    size_t offset = 0;
+    for (int i = 1; i < size; i++) {
+        cpu_memcpy(out + offset * elemSize, in + i * size * elemSize, i * elemSize);
+        offset += i;
+    }
+}
+
+void Interaction::execRef(dnnl::stream strm) {
+    using tag = dnnl::memory::format_tag;
+    using dt = dnnl::memory::data_type;
+    using namespace dnnl;
+
+    uint8_t* outFeaturesPtr = reinterpret_cast<uint8_t*>(getChildEdgesAtPort(0)[0]->getMemoryPtr()->GetPtr());
+    std::vector<const uint8_t*> inputPtrs(inputSizes);
+    for (uint32_t n = 0; n < inputSizes; n++) {
+        auto inPtr = reinterpret_cast<const uint8_t*>(getParentEdgeAt(n)->getMemoryPtr()->GetPtr());
+        inputPtrs[n] = inPtr;
+    }
+    for (int64_t start = 0; start < batchSize; start++) {
+        cat(inputPtr->buffer().as<uint8_t*>(), inputPtrs, featureSizes, start, dataPrecision.size());
+        std::unordered_map<int, memory> mem_ags {
+            {DNNL_ARG_SRC, inputMemPtr->GetPrimitive()},
+            {DNNL_ARG_WEIGHTS, inputMemPtr->GetPrimitive()},
+            {DNNL_ARG_DST, outputMemPtr->GetPrimitive()}
+        };
+        (*prim).execute(strm, mem_ags);
+        flat_triangle(outputPtr->buffer().as<uint8_t*>(),
+            flatPtr->buffer().as<uint8_t*>(), inputSizes, dataPrecision.size());
+        //in1 dense feature
+        //in2 flatted interaction features
+        cat(
+          inputPtrs[0] + start * featureSize * dataPrecision.size(),
+          flatPtr->buffer().as<uint8_t*>(),
+          outFeaturesPtr + start * outputFeaturesLen * dataPrecision.size(),
+          featureSize,
+          interactFeatureSize,
+          dataPrecision.size());
+    }
+}
+
+
+
+void Interaction::execute(dnnl::stream strm) {
+    execRef(strm);
+}
+
+bool Interaction::created() const {
+    return getType() == Type::Interaction;
+}
+
+void Interaction::prepareParams() {
```

Comment:
It is really nice that the impl supports dynamism. My concern is performance: I don't see that primitive caching is used here. So there is no guarantee that Interaction op provides better performance in case of dynamic shapes comparing with decomposed subgraph considering recompilation time. 
What I can suggest:

- If we have time lets add caching support and do proper benchmarking in dynamic shapes scenario. 
- If no lets disable dynamic case inside ConvertToInteraction transformation

Reply:
I prefer the temporarily disable dynamic shape case. The dynamic shape support is added upon the review comments. I think it's not well support

Reply:
disable dynamic check temporarily in e7c719fd049b5a1c00796112fb84e6753b3cdb3c